client/src/Widgets/ActivityTable.tsx

```diff
@@ -0,0 +1,242 @@
+
+export default forwardRef(function ActivityTable(
+  {
+    content,
+  }: {
+    content: {
+      cardLink?: string;
+      id: number;
+      assignmentStatus?: AssignmentStatus;
+      isFolder?: boolean;
+      isPublic?: boolean;
+      title: string;
+      ownerName?: string;
+      menuItems?: ReactElement;
+      suppressAvatar?: boolean;
+      showOwnerName?: boolean;
+      editableTitle?: boolean;
+      autoFocusTitle?: boolean;
+      showAssignmentStatus?: boolean;
+      showPublicStatus?: boolean;
+      closeTime?: string;
+    }[];
+  },
+  ref: React.ForwardedRef<HTMLButtonElement>,
+) {
+  const navigate = useNavigate();
+  const fetcher = useFetcher();
+
+  return (
+    <TableContainer>
+      <Table>
+        <Thead>
+          <Tr borderBottom="2px solid gray">
+            <Th></Th>
+            <Th></Th>
+            {content.some((act) => act.showPublicStatus) ? (
+              <Th>Visibility</Th>
+            ) : (
+              <Th></Th>
+            )}
+            {content.some((act) => act.showAssignmentStatus) ? (
+              <Th>Assignment Status</Th>
+            ) : (
+              <Th></Th>
+            )}
+            {content.some((act) => act.showOwnerName || !act.suppressAvatar) ? (
+              <Th>Owner</Th>
+            ) : (
+              <Th></Th>
+            )}
+            <Th></Th>
+          </Tr>
+        </Thead>
+        <Tbody>
+          {content.map(function (activity) {
+            if (activity.isFolder) {
+              activity.showAssignmentStatus = false;
+            }
+
+            let assignmentStatusString: string =
+              activity.assignmentStatus !== null &&
+              activity.assignmentStatus !== undefined &&
+              activity.assignmentStatus !== "Unassigned"
+                ? activity.assignmentStatus
+                : "";
+
+            if (
+              activity.assignmentStatus === "Open" &&
+              activity.closeTime !== undefined
+            ) {
+              assignmentStatusString =
+                assignmentStatusString + " until " + activity.closeTime;
+            }
+
+            function saveUpdatedTitle(newTitle: string) {
+              if (newTitle !== activity.title && activity.id !== undefined) {
+                fetcher.submit(
+                  {
+                    _action: "update title",
+                    id: activity.id,
+                    cardTitle: newTitle,
+                    isFolder: Boolean(activity.isFolder),
+                  },
+                  { method: "post" },
+                );
+              }
+            }
+
+            return (
+              <Tr
+                key={activity.id + "_row"}
+                cursor="pointer"
+                _hover={{ backgroundColor: "#eeeeee" }}
+                borderBottom="2px solid gray"
+                onClick={() =>
+                  activity.cardLink ? navigate(activity.cardLink) : null
```

Comment:
Just missing the menu when clicking or selecting a menu item still triggers this navigation. You could probably make the menu item margin or padding bigger to make it require less precise clicking. I'm not sure what do do about the menu items triggering the navigation.

If I remember right, I had this trouble when trying to make the full activity card clickable, and ended up giving up for the case where there was a menu. If you run into the same trouble, you could always attach the `onClick` listener to something smaller for the case with a menu (i.e., for Activities as opposed to Community).